cloud/openstack/os_user_role.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python
+# Copyright (c) 2015 Hewlett-Packard Development Company, L.P.
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+
+try:
+    import shade
+    from shade import meta
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_user_role
+short_description: Associate OpenStack Identity users and roles
+extends_documentation_fragment: openstack
+version_added: "2.0"
+description:
+    - Grant and revoke roles in either project or domain context for
+      OpenStack Identity Users
+options:
+   role:
+     description:
+        - Name or id for the role
+     required: true
+   user:
+     description:
+        - Name or id for the user. (user or group required)
+     required: true
+   group:
+     description:
+        - Name or id for the group. (user or group required)
+     required: true
+   project:
+     description:
+        - Name or id of the project to scope the role assocation to.
+     required: false
+     default: None
+   domain:
+     description:
+        - Name or id of the domain to scope the role association to.
+     required: false
+     default: None
+   state:
+     description:
+       - Should the roles be present or absent on the user
+     choices: [present, absent]
+     default: present
+requirements:
+    - "python >= 2.6"
+    - "shade"
+'''
+
+EXAMPLES = '''
+# Grant an admin role on the user admin in the project project1
+- os_user_role: user=admin role=admin project=project1 domain=default
+# Revoke the awesome role from the user barney in the newyork domain
+- os_user_role: user=barney role=awesome domain=newyork state=absent
+'''
+
+
+def main():
+
+    argument_spec = openstack_full_argument_spec(
+    argument_spec = dict(
+        role=dict(required=True),
+        user=dict(required=True),
+        group=dict(required=True),
```

Comment:
Docs say that user /or/ group are required, but both are listed as required here. Which is right?